src/schoology-extractor/schoology_extractor/main.py

```diff
@@ -144,6 +147,16 @@ def __get_attendance_events() -> pd.DataFrame:
     return __get_attendance_events
 
 
+def _get_system_activities(input_directory: str) -> Callable:
```

Comment:
`input_directory` is a class variable, so there's really no need to pass it to the function. And because you don't need to pass it to the function, you don't need to create closure over it - so you could eliminate the inner function.

Reply:
✔️